src/package-linker.js

```diff
@@ -409,16 +409,17 @@ export default class PackageLinker {
 
     for (const name in peerDeps) {
       const range = peerDeps[name];
-      const patterns = this.resolver.patternsByPackage[name] || [];
-      const foundPattern = patterns.find(pattern => {
-        const resolvedPattern = this.resolver.getResolvedPattern(pattern);
-        return resolvedPattern ? this._satisfiesPeerDependency(range, resolvedPattern.version) : false;
+      const pkgs = this.resolver.getAllInfoForPackageName(name);
+      const found = pkgs.find(pkg => {
+        const {root, version} = pkg._reference || {};
+        return root && this._satisfiesPeerDependency(range, version);
       });
+      const foundPattern = found && found._reference && found._reference.patterns;
```

Comment:
Not sure if this is you or flow yelling at you but you don't need to check `found._reference` since it is required for `.find()` to return something for `found`. Actually you can even move `patterns` up there and just check `found` as I suggested earlier. I'm guessing you did it this way because of `flow` complaining tho :D